modules/apps/27-interchain-accounts/controller/keeper/export_test.go

```diff
@@ -0,0 +1,10 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// SetParams sets the total set of params.
+func (k Keeper) HasCapability(ctx sdk.Context, portID string) bool {
```

Comment:
@expertdicer Is this public function really needed?

Reply:
Yes. It's for separating testing logic and core private logic. So in the future, if you want to modify in the testing layer, it won't affect the core private logic. I think so.

Reply:
Seems like the documentation for this refers to something else?

Reply:
> Seems like the documentation for this refers to something else?

Sorry. I wasn't checking the comment carefully.

Reply:
This is a really cool construction. Never thought about wrapping private functions with a public testing only API. I do think I'd prefer if we removed these functions if they are unused. We can add them back if we find usage for them, but sometimes it's good to write tests in the same way external users will need to access information

Reply:
I agree with @colin-axner that if the function is not used at the moment, then better to just remove it for now.